_posts/2022-08-10-#25720.md

```diff
@@ -10,15 +10,53 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- On startup, if a node's headers chain is more than a day behind the current
+  time, it will pick one peer to sync headers with until its best headers chain
+  is caught up. (See
+  [here](https://github.com/bitcoin/bitcoin/blob/92f6461cfd39fff2fc885dd623fa47e7d8d53827/src/net_processing.cpp#L4889-L4913))
+  If a node's headers chain is not caught up within `15 min + ~num headers
+  missing * 1ms` (either because the chosen peer was not able to or because
+  they purposefully stalled the download), it will disconnect the chosen peer
+  for stalling and pick a new peer to sync headers with.
+
+- When a node receives a block announcement via an INV during headers sync, it
+  will add all announcing peers as additional headers sync peers. This week's
+  PR changes this logic such that only one of the announcing peers is added for
+  headers sync.
+
+- [BIP 130](https://github.com/bitcoin/bips/blob/master/bip-0130.mediawiki)
+  (implemented in Bitcoin Core since late 2015) introduced the `sendheaders`
+  message, which indicates that a node prefers receiving block announcements
+  via a `headers` message rather than an `inv`.
+
 ## Questions
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
 
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+
+2. Why do nodes (mostly) receive `inv` block announcements while they are doing
+   initial headers sync, even though they indicated preference for `headers`
+   announcements (BIP 130)?
+
+3. Why is bandwidth wasted (during initial headers sync) by adding all peers
+   that announce a block to us via an INV as headers sync peers?
+
+4. What would your estimate (lower/upper bound) be of how much bandwidth is
+   wasted? (There is no one true answer here, the estimate depends on a couple
+   of variables)
+
+5. If we start syncing headers with peers that announce a block to us via an
+   INV, why do we not increase `nSyncStarted` and set `fSyncStarted = true` and
+   `m_headers_sync_timeout`? (Please start by explaining what these variables
+   are used for)
```

Comment:
I think it's an easier format (especially live in the review club) to break that up so you can ask the first question first, e.g.:

What's the purpose of `CNodeState`'s members `fSyncStarted` and `m_headers_sync_timeout`, and `PeerManagerImpl::nSyncStarted`? If we start syncing headers with peers that announce a block to us via an
   INV, why do we not increase `nSyncStarted` and set `fSyncStarted = true` and update
   `m_headers_sync_timeout`?